sqlobject/src/main/java/org/jdbi/v3/sqlobject/statement/internal/ResultReturner.java

```diff
@@ -57,7 +60,7 @@ static ResultReturner forOptionalReturn(Class<?> extensionType, Method method) {
      * @return an instance that takes a ResultIterable and constructs the return value
      */
     static ResultReturner forMethod(Class<?> extensionType, Method method) {
-        Type returnType = GenericTypes.resolveType(method.getGenericReturnType(), extensionType);
+        Type returnType = GenericTypeReflector.getReturnType(method, extensionType);
```

Comment:
If this can't keep using `GenericTypes.resolveType()` then this is a breaking change.